enhancements/proposals/NNNN-alternator/README.md

```diff
@@ -0,0 +1,164 @@
+
+## Design Details
+
+### Approximate API
+
+```golang
+package v1
+
+type TLSCertificateType string
+
+const (
+	ManagedCertificateType     TLSCertificateType = "OperatorManaged"
+	UserManagedCertificateType TLSCertificateType = "UserManaged"
+)
+
+type UserManagedTLSCertificateOptions struct {
+	// secretName references a kubernetes.io/tls type secret containing the TLS cert and key.
+	// +kubebuilder:validation:Required
+	SecretName string `json:"secretName"`
+}
+
+type OperatorManagedTLSCertificateOptions struct {
+	// additionalDNSNames represents external DNS names that the default certificates should be signed for.
+	// +optional
+	AdditionalDNSNames []string `json:"additionalDNSNames,omitempty"`
+	
+	// additionalIPAddresses represents external DNS names that the default certificates should be signed for.
+	// +optional
+	AdditionalIPAddresses []string `json:"additionalIPAddresses,omitempty"`
+}
+
+type TLSCertificate struct {
+	// name is the metadata.name of the referenced secret
+	// +kubebuilder:validation:Enum="OperatorManaged";"UserManaged"
+	Type TLSCertificateType `json:"type"`
+	
+	// userManagedCertificateOptions specify option for certificates manged by users.
+	// +optional
+	UserManagedOptions *UserManagedTLSCertificateOptions `json:"userManagedOptions,omitempty"`
+	
+	// OperatorManagedCertificateOptions specify option for certificates manged by the operator.
+	// +optional
+	OperatorManagedOptions *OperatorManagedTLSCertificateOptions `json:"operatorManagedOptions,omitempty"`
+}
+
+// Only new fields are shown in the alternator spec. 
+
+type AlternatorSpec struct {
+	// insecureEnableHTTP enables serving Alternator traffic also on insecure HTTP port.
+	// (This needs to default to false but only for CREATE, not UPDATE - so we need a webhook instead of the kubebuilder default)
+	InsecureEnableHTTP *bool `json:"insecureEnableHTTP,omitempty"`
+	
+	// servingCertificate references a TLS certificate for serving secure traffic.
+	// +kubebuilder:default:="{type:"OperatorManaged"}"
+	// +optional
+	ServingCertificate *TLSCertificate `json:"servingCertificate,omitempty"`
+}
+```
+
+### Certificates
+
+Given we aim to always enable HTTPS we need to provision default serving certificate. Operator will create signer Secret `<sc-name>-local-alternator-serving-ca`, save the CA bundle into ConfigMap `<sc-name>-local-alternator-serving-ca` and create serving Secret `<sc-name>-local-alternator-serving-certs`.
+Serving certs will be valid for 30 days and rotated from 20 day mark.
+Because users can create an alternative Service for the alternator (e.g. LB type), we'll allow specifying additional DNS names and IPs to include in the managed certificates. 
+
+Optionally, users can decide they want to provide a serving certificate on their own, which will disable the operator provided certificates.
+This is useful in case users want the service to have a certificate signed by a CA in their root of trust or for similar customizations.
+
```

Comment:
Details paragraph doesn't cover anything other than certificates and API while entire feature touches different areas. Please describe new webhook validation rules, which Scylla configuration keys are managed, what is the source of values for these configuration entries. Whether users override these by specifying them via their CM, how to protect from it.

Reply:
added details